framework-plugins/lisk-framework-chain-connector-plugin/test/unit/utils.spec.ts

```diff
@@ -12,92 +12,134 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { ActiveValidatorWithAddress } from '../../src/types';
-import { getActiveValidatorsDiff } from '../../src/active_validators_update';
+import { getActiveValidatorsUpdate } from '../../src/active_validators_update';
 
-describe('getActiveValidatorsDiff', () => {
-	it('should aggregate new validators not existing in existing validators', () => {
-		const currentValidators: ActiveValidatorWithAddress[] = [
-			{
-				blsKey: Buffer.from('02', 'hex'),
-				bftWeight: BigInt(20),
-				address: Buffer.from('bb', 'hex'),
-			},
-		];
-
-		const newValidators: ActiveValidatorWithAddress[] = [
-			{
-				blsKey: Buffer.from('02', 'hex'),
-				bftWeight: BigInt(20),
-				address: Buffer.from('bb', 'hex'),
-			},
-			{
-				blsKey: Buffer.from('03', 'hex'),
-				bftWeight: BigInt(30),
-				address: Buffer.from('cc', 'hex'),
-			},
-			{
-				blsKey: Buffer.from('04', 'hex'),
-				bftWeight: BigInt(40),
-				address: Buffer.from('dd', 'hex'),
-			},
-		];
-
-		const expectedValidators: ActiveValidatorWithAddress[] = [
-			{
-				blsKey: Buffer.from('03', 'hex'),
-				bftWeight: BigInt(30),
-				address: Buffer.from('cc', 'hex'),
-			},
-			{
-				blsKey: Buffer.from('04', 'hex'),
-				bftWeight: BigInt(40),
-				address: Buffer.from('dd', 'hex'),
-			},
-		];
-
-		const actualValidators = getActiveValidatorsDiff(currentValidators, newValidators);
-
-		expect(actualValidators).toStrictEqual(expectedValidators);
-	});
+describe('getActiveValidatorsUpdate', () => {
+	const bytesToBuffer = (str: string): Buffer => {
+		const val = BigInt(`0b${str}`).toString(16);
+		return Buffer.from(`${val.length % 2 === 0 ? val : `0${val}`}`, 'hex');
+	};
 
-	it('should aggregate existing validators having blsKey not existing in new validators with bftWeight set to 0 and aggregating the new validator as well', () => {
-		const currentValidators: ActiveValidatorWithAddress[] = [
+	const cases = [
```

Comment:
These test cases look good. I would still add a test case to just remove some validators without adding any and one to remove and change BFT weights of others. Also duplicate all tests to also add an extra validator which is left untouched.